### DISCUSSION TITLE
TG799vac Xtream doesn't boot, how to read bootlog from serial console?

### DISCUSSION THREAD

**Discussion by frehall** [hack-technicolor/hack-technicolor#129] [Q&A]
After i had my router rooted i ran [https://raw.githubusercontent.com/nutterpc/tg-luci/master/install.sh](https://raw.githubusercontent.com/nutterpc/tg-luci/master/install.sh)  to get LuCi and after that it does not boot. I can upload new .rbi image with tftpd but it does not boot. Also ive been trying to get serial console following [this](https://github.com/hack-technicolor/hack-technicolor/issues/79) thread but with no success. I think that thread is a little unclear in what i have to do.. how to connect and if i need to solder or not?

**Comment by LuKePicci** [comment]
>  I can upload new .rbi image with tftpd but it does not boot.

Did you apply optimal bank plan? If no, and no firmware is inside bank_2 then this behavior is expected.

> how to connect

find a similar 4 pin header on the board, the pinout is often the same, take a picture, identify GND pin and TX. RX isn't needed if you only need to read the bootlog.

> if i need to solder or not?

it depends on your board and what you need to do, I usually don't solder for a quick bootlog reading it's enough to touch the two above mentioned contacts with wires.

You can read more about serial console here: https://openwrt.org/docs/techref/hardware/port.serial or google about that, the story is often the same.

**Comment by frehall** [comment]
> Did you apply optimal bank plan? If no, and no firmware is inside bank_2 then this behavior is expected.

I did not.

> find a similar 4 pin header on the board, the pinout is often the same, take a picture, identify GND pin and TX. RX isn't needed if you only need to read the bootlog.

i have the same board as [https://user-images.githubusercontent.com/8722358/78917493-35d0a680-7a8f-11ea-9824-48555b30d161.jpg](https://user-images.githubusercontent.com/8722358/78917493-35d0a680-7a8f-11ea-9824-48555b30d161.jpg) but i dont get any output. i think its kind of unclear how i should wire it up, can you help me on that one please? also maybe im using wrong stuff to get serial going? Termite program and chip: [https://www.amazon.com/DSD-TECH-Adapter-FT232RL-Compatible/dp/B07BBPX8B8](https://www.amazon.com/DSD-TECH-Adapter-FT232RL-Compatible/dp/B07BBPX8B8)

**Comment by LuKePicci** [comment]
Adapter is fine, make sure to set its jumper in 3.3V mode - not 5V

Also make sure you're connecting to the red-circled SoC serial header in the picture, and not the cyan Quantenna one.

The wiring is:
adapter TX <--> board RX : yellow cable [not required for read-only operations]
adapter RX <--> board TX : green cable
adapter GND <> board GND : black cable

Above colors refer to cables in this picture: https://github.com/hack-technicolor/hack-technicolor/issues/79#issuecomment-583882093

You only need to wire the two in the middle for read-only console.

**Comment by frehall** [comment]
Ok, this is what im trying to do. Or at least i think so. Ill upload an image of my exact board:

[https://pasteboard.co/JymuxfO.jpg](https://pasteboard.co/JymuxfO.jpg)

I dont get anyoutput to console :(

**Comment by LuKePicci** [comment]
Check serial adapter settings from PC, 

![image](https://user-images.githubusercontent.com/8722358/97807653-5302d780-1c62-11eb-87f7-c3ba31d0017f.png)

**Comment by frehall** [comment]
Finaly i had a breakthrough and was able to connect to serial... The complete log is [here](https://paste.debian.net/1169523/) maybe someone can help me understand whats wrong and if theres anything I can do to make it productive again?

Update: Ive managed to switch to bank_2 as in [this video](https://www.youtube.com/watch?v=BMT8AhA4qns) and boot that. From there I can root the box and enable ssh for root at LAN. For some reason the box reboots after a while and returns to bank_1.
Guess I have to fix broken bank_1 in some way.

**Comment by LuKePicci** [comment]
So, disconnect any wan cable , reboot into bank_2, enter ssh and run "mtd -r erase rootfs_data". This will clean up both banks.

Also, RTFD via reset button would fix bank_1 while it's booted, but I'm not sure RTFD is not getting broken by the bad libuci package you installed..

**Comment by frehall** [comment]
Ok, that seems to be working allright.

So, to wrap this up a bit. 
I messed up my router running some script to get LuCi on it. Then I  couldnt flash it with TFTPD as none of the rbi:s listed for my ISP (Telia .se) was accepted by the router. Had some problems to get a good serial connection and had to do some soldering. Finaly connected to serial i could learn that my router took alot more time to boot than in the bankswitching-video linked above here. So when I had serial I was able to time the powerflicking and get bank_2 booted. Rooted it, and ran "mtd -r erase rootfs_data".

Too bad none of the repos seems to work with the router but at least it can operate as a switch. :)

**Comment by LuKePicci** [comment]
These firmware images are expected to be accepted on your VBNT-H board https://hack-technicolor.rtfd.io/en/latest/Repository/#telia-wifi-router-plus-v3
You can follow signature verification guide from the wiki in order to confirm this without actually flashing anything, if you like. In the strange case they do not pass signcheck from your board then I would ask you to dump your board keys and firmware and share them so we could check.

Correct package repo for your device exists and depends on fw version. Open a different issues for that if you're interested.
